src/lib/ethers/zksync/server.ts

```diff
@@ -0,0 +1,291 @@
+
+/**
+ * Leverages `JsonRpcEngine` to intercept account-related calls, and pass any other calls down to a destination
+ * provider, e.g. Infura.
+ */
+class WalletMiddlewareServer {
+  alwaysSynced: boolean
+  expressServer: Express
+  mockFilters: boolean
+  privateKeys?: string[]
+  seedPhrase?: string
+  seedPhraseWallets: number
+  wrapper: WalletWrapper
+
+  constructor (
+    seed_phrase: string,
+    seed_phrase_wallets: number,
+    private_keys: string[],
+    interleave_blocks: number,
+    gas_price: number,
+    gas_limit: number,
+    estimate_gas_limit: boolean,
+    estimate_gas_price: boolean,
+    always_synced: boolean,
+    mock_filters: boolean,
+    gas_price_factor: number,
+    gas_limit_factor: number,
+    force_eip_155: boolean,
+    eth_gas_price_factor: boolean,
+    provider?: Provider
+  ) {
+    this.alwaysSynced = always_synced
+    this.expressServer = express()
+    this.mockFilters = mock_filters
+    this.privateKeys = private_keys
+    this.seedPhrase = seed_phrase
+    this.seedPhraseWallets = seed_phrase_wallets
+    this.wrapper = new WalletWrapper(
+      interleave_blocks,
+      gas_price,
+      gas_limit,
+      estimate_gas_limit,
+      estimate_gas_price,
+      gas_price_factor,
+      gas_limit_factor,
+      force_eip_155,
+      eth_gas_price_factor
+    )
+    if (provider) {
+      this.wrapper.provider = provider
+    }
+    // trace configuration
+    let lines = [
+      [
+        'Gas Price',
+        `${gas_price} ${estimate_gas_price ? '(max)' : '(default)'}`
+      ],
+      [
+        'Gas limit',
+        `${gas_limit} ${estimate_gas_limit ? '(max)' : '(default)'}`
+      ]
+    ]
+    if (gas_price_factor > 1) {
+      lines = [...lines, ['Gas price factor', `x ${gas_price_factor}`]]
+    }
+    if (gas_limit_factor > 1) {
+      lines = [...lines, ['Gas limit factor', `x ${gas_limit_factor}`]]
+    }
+    if (interleave_blocks > 0) {
+      lines = [...lines, ['Interleave blocks', interleave_blocks.toString()]]
+    }
+    traceKeyValue('Config', lines)
+    return this
+  }
+
+  /**
+   * Initializes the Express server, configures CORS, and passes requests back and forth between the Express server and
+   * the `JsonRpcEngine`.
+   */
+  initialize () {
+    this.expressServer.use(cors())
+    this.expressServer.use(express.json())
+
+    this.expressServer.post(
+      '*',
+      async (req: express.Request, res: express.Response) => {
+        const request = req.body
+        const socket = {
+          clientAddr: req.connection.remoteAddress,
+          clientPort: req.connection.remotePort,
+          clientId: request.id,
+          serverId: this.wrapper.provider._nextId
+        }
+
+        logger.log({
+          level: 'info',
+          socket,
+          message: `>> ${request.method}`
+        })
+
+        let handlers: { [K: string]: any } = {
+          eth_accounts: this.wrapper.getAccounts,
+          // eth_call: this.wrapper.processEthCall,
```

Comment:
Are this methods commented because they aren't supported or because we don't need them now but we'll need them in the future to add support for price feeds?

Reply:
`eth_*` methods not explicitly included in the handlers map are not intercepted and therefore natively supported by the provider.

Reply:
Once tested and proven to be indeed supported, commented lines will now be removed.